src/netlink.c

```diff
@@ -0,0 +1,365 @@
+    }
+
+    lua_pushinteger(L, got);
+    lua_pushlstring(L, NLMSG_DATA(nlh), got);
+    lua_pushinteger(L, nlh->nlmsg_pid);
+    return 3;
+}
+
+/*-------------------------------------------------------------------------*\
+* Select support methods
+\*-------------------------------------------------------------------------*/
+static int meth_getfd(lua_State *L) {
+    p_netlink nl = (p_netlink)auxiliar_checkgroup(L, "netlink{any}", 1);
+    lua_pushnumber(L, (int)nl->fd);
+    return 1;
+}
+
+/* this is very dangerous, but can be handy for those that are brave enough */
+static int meth_setfd(lua_State *L) {
+    p_netlink nl = (p_netlink)auxiliar_checkgroup(L, "netlink{any}", 1);
+    nl->fd = (t_socket)luaL_checknumber(L, 2);
+    return 0;
+}
+
+static int meth_getsockpid(lua_State *L) {
+    p_netlink nl = (p_netlink)auxiliar_checkgroup(L, "netlink{any}", 1);
+    lua_pushinteger(L, nl->srcpid);
+    return 1;
+}
+
+/*-------------------------------------------------------------------------*\
+* Binds an object to an address
+\*-------------------------------------------------------------------------*/
+static const char *netlink_trybind(p_netlink nl, int grp) {
+    struct sockaddr_nl addr;
+    int err;
+
+    memset(&addr, 0, sizeof(addr));
+    addr.nl_family = AF_NETLINK;
+    addr.nl_groups = grp;	
+    addr.nl_pid = nl->srcpid;
+    getpeername(nl->fd, (SA *)&addr, (socklen_t *)sizeof(addr));
+    err = socket_bind(&nl->fd, (SA *)&addr, sizeof(addr));
+    return socket_strerror(err);
+}
+
+static int meth_bind(lua_State *L) {
+    int pid = luaL_checkinteger(L, 2);
+    int grp = luaL_optinteger(L, 3, 0);
+    const char *err;
+    p_netlink nl = (p_netlink)auxiliar_checkclass(L, "netlink{unconnected}", 1);
+
+    nl->srcpid = pid;
+    err = netlink_trybind(nl, grp);
+    if (err != NULL) {
+        lua_pushnil(L);
+        lua_pushstring(L, err);
+        return 2;
+    }
+
+    lua_pushnumber(L, 1);
+    return 1;
+}
+
+
+/*-------------------------------------------------------------------------*\
+* Closes socket used by object
+\*-------------------------------------------------------------------------*/
+static int meth_close(lua_State *L) {
+    p_netlink nl = (p_netlink)auxiliar_checkgroup(L, "netlink{any}", 1);
+    socket_destroy(&nl->fd);
+    lua_pushnumber(L, 1);
+    return 1;
+}
+
+/*-------------------------------------------------------------------------*\
+* Just call peername methods
+\*-------------------------------------------------------------------------*/
+static int meth_setpeername(lua_State *L) {
+    p_netlink nl = (p_netlink)auxiliar_checkgroup(L, "netlink{any}", 1);
+    struct sockaddr_nl addr;
+    int dstpid;
+    int grps;
+    int err;
+
+    if (lua_isnone(L, 2)) {
+        addr.nl_family = AF_UNSPEC;
+        socket_connect(&nl->fd, (SA *)&addr, 
+                (socklen_t)sizeof(addr), &nl->tm);
+        auxiliar_setclass(L, "netlink{unconnected}", 1);
+        return 0;
+    }
```

Comment:
Where did this came from? Do you have links for documentation that show this is valid and what is means exactly?

Reply:
It came from Diego's mind pretty much. It's an abstraction of udp sockets, since, by definition, they 'cannot' be in a connected or unconnected state. Connecting them only means that the destination port id is stored internally, so that the user can use the send function without having to specify the destination port everytime. Although, theoretically, I believe it's not the most correct thing.

Reply:
I'm really not interested in your mind reading powers, as mush impressive they might seem, so how about you pointing me the original Diego's code that use based on to write this, and more importantly which documentation about Netlink made you thing it is correct to pass a `AF_UNSPEC` to `connect` and what does it mean.

Reply:
If you wanted to ask about AF_UNSPEC, perhaps you should've added your comment on line 284 instead of line 289, because even though I can read minds, I guess I can't read yours. You asked "where did this come from" and pointed to a closing curly bracket...The only section of code you could be asking about that showed in the PR comment section was the line with "netlink{unconnected}", so that's what I explained to you.. I only ask you to not be so condescending next time, it's unhealthy (Wouldn't it be better if you just replied, "I know what that means, I was asking about AF_UNSPEC"?). Now to answer your question, I took that from the man page of 'connect'
> Generally, connection-based protocol sockets may successfully connect()
       only once; connectionless protocol sockets may use  connect()  multiple
       times to change their association.  Connectionless sockets may dissolve
       the association by connecting to an address with the  sa_family  member
       of sockaddr set to AF_UNSPEC (supported on Linux since kernel 2.2).